warehouse/warehouse.go

```diff
@@ -1230,6 +1263,12 @@ func (wh *HandleT) Setup(whType string) {
 	config.RegisterIntConfigVariable(1, &wh.maxConcurrentUploadJobs, false, 1, fmt.Sprintf(`Warehouse.%v.maxConcurrentUploadJobs`, whName))
 	config.RegisterBoolConfigVariable(false, &wh.allowMultipleSourcesForJobsPickup, false, fmt.Sprintf(`Warehouse.%v.allowMultipleSourcesForJobsPickup`, whName))
 
+	wh.internalAuth = cpclient.BasicAuth{
+		Username: config.GetString("CP_INTERNAL_API_USERNAME", ""),
+		Password: config.GetString("CP_INTERNAL_API_PASSWORD", ""),
+	}
+	wh.configBackendURL = configBackendURL
```

Comment:
```suggestion

       wh.CPInternalClient =  cpclient.NewInternalClientWithCache(configBackendURL, cpclient.BasicAuth{
		Username: config.GetString("CP_INTERNAL_API_USERNAME", ""),
		Password: config.GetString("CP_INTERNAL_API_PASSWORD", ""),
	})
```

It is preferable to DI the client rather than the config/credentials; it would be easier to mock the client in the feature.

However, it might be even better to inject the config middleware here instead depending on the client directly.